src/plot_api/plot_api.js

```diff
@@ -941,6 +941,7 @@ Plotly.redraw = function(gd) {
 Plotly.newPlot = function (gd, data, layout, config) {
     gd = getGraphDiv(gd);
     plots.purge(gd);
+    gd.removeAllListeners();
```

Comment:
Good call on moving `removeAllListeners`. Not sure if we should expose the purge method though - I can't think of too many use cases where that's what a user would want to clear the node and reuse it - if they want to "purge", they'll just delete the node. Better to keep the user surface area of plotly.js as small as possible. 

@alexcjohnson thoughts?